src/nyc_trees/apps/survey/management/commands/treecorder.py

```diff
@@ -0,0 +1,148 @@
+    '+lat_0=40.16666666666666 +lon_0=-74 +x_0=300000 +y_0=0 '
+    '+ellps=GRS80 +datum=NAD83 +to_meter=0.3048006096012192 +no_defs ')
+ny_state_plane.validate()
+
+wgs84 = SpatialReference(4326)
+
+ny_to_latlng = CoordTransform(ny_state_plane, wgs84)
+
+
+class Command(BaseCommand):
+    """
+    Execute the Treekit geometry constructor for each row of tree data
+    in a CSV file. The output CSV should match the original input with
+    additional columns for lat/lng.
+    """
+
+    def handle(self, *args, **options):
+        fp = fileinput.input('-')
+        reader = csv.reader(fp)
+        writer = csv.writer(sys.stdout)
+
+        headers = reader.next()
+        headers = headers + ['LAT', 'LNG']
+        writer.writerow(headers)
+
+        records = (Record(*line) for line in reader)
+
+        # Group rows in chunks by block_face_number. Assumes that the CSV
+        # is sorted by block_face_number.
+        grouped_records = itertools.groupby(records,
+                                            lambda r: r.block_face_number)
+
+        for blockface_id, rows in grouped_records:
+            # The WITH_XY spreadsheet has a bunch of duplicate survey rows,
+            # need to deduplicate...
+            seen = set()
+            rows = [row for row in rows
+                    if row.tree_id_number not in seen and
+                    not seen.add(row.tree_id_number)]
+            survey = self.create_survey(rows[0])
+
+            if survey is None:
+                continue
+
+            trees = list(self.create_trees(rows))
+
+            # Execute Treekit geometry constructor to determine
+            # tree locations.
+            points = _survey_geojson(survey, trees)
+
+            # Parse GeoJSON result.
+            points = (json.loads(p) for p in points)
+
+            # Combine tree location with original CSV row.
+            for row, point in itertools.izip(rows, points):
+                lat, lng = point['coordinates']
+                writer.writerow(row + (lat, lng))
+
+    def create_survey(self, row):
+        """
+        Return fake survey with the minimum amount of information
+        necessary to generate tree locations.
+        """
+        survey = Survey()
+        survey.blockface_id = row.block_face_number
+        survey.is_left_side = row.side_of_centerline == 'Left'
+
+        try:
+            blockface = Blockface.objects.get(pk=survey.blockface_id)
+        except Blockface.DoesNotExist:
+            return None
+
+        # We need to pass an srid to call .transform(), but it is ignored if we
+        # give it a CoordTransform object
+        survey_start_point = Point(float(row.x_point), float(row.y_point),
+                                   srid=102718)
+        survey_start_point.transform(ny_to_latlng)
+
+        blockface_start = Point(blockface.geom.coords[0][0], srid=4326)
+        blockface_end = Point(blockface.geom.coords[0][0], srid=4326)
```

Comment:
Doh!

I changed this to be `Point(blockface.geom.coords[0][-1], srid=4326)` (the last point of the first segment) in  401beb2.

I'm not sure what blockfaces (if any) have multiple line segments, but the [`getLatLngs` in `mapUtils.js`](https://github.com/azavea/nyc-trees/blob/develop/src/nyc_trees/js/src/lib/mapUtil.js#L21-L28) always uses the first segment, so I chose to mirror that.